menuinst/platforms/win_utils/knownfolders.py

```diff
@@ -313,3 +316,35 @@ def folder_path(preferred_mode, check_other_mode, key):
         )
         return None
     return path
+
+
+def windows_terminal_settings_files(mode: str) -> List[Path]:
+    """Return all possible locations of the settings.json files for the Windows Terminal.
+
+    See the Microsoft documentation for details:
+    https://learn.microsoft.com/en-us/windows/terminal/install#settings-json-file
+    """
+    if mode != "user":
+        return []
+    localappdata = folder_path(mode, False, "localappdata")
+    packages = Path(localappdata) / "Packages"
+    profile_locations = [
+        # Stable
+        *[
+            Path(terminal, "LocalState", "settings.json")
+            for terminal in packages.glob("Microsoft.WindowsTerminal_*")
+        ],
+        # Preview
+        *[
+            Path(terminal, "LocalState", "settings.json")
+            for terminal in packages.glob("Microsoft.WindowsTerminalPreview_*")
+        ],
+        # Unpackaged (Scoop, Chocolatey, etc.)
+        Path(
+            localappdata,
+            "Microsoft",
+            "Windows Terminal",
+            "settings.json",
+        ),
+    ]
+    return profile_locations
```

Comment:
Are we not checking for existence here?

Reply:
My initial thought was, when the folders are still hard-coded, that these are the "known folders", i.e., where these files could be. However, I can also just check for existence right here, that's true.

In that case, I could also change the tests so that they just create a directory in the tests if no directory is found. That way, it's at least testing something. Does that sound good?

Reply:
Yep!

Reply:
Since the terminal doesn't appear to be CI-installable, I decided to do away with all removing the location mocking and just use a mock directory for the terminal test.

I ran into another problem though: on the CI, you are always admin, and the terminal has no definition for an admin profile. So, I had to mock the `user_is_admin` function to return `False` for this test.